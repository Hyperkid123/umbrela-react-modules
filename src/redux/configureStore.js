import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
