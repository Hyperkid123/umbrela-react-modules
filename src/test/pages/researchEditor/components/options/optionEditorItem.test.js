import React, {Component} from 'react'
import OptionEditorItem, {OptionEditorItem as Snapshot} from '../../../../../pages/researchEditorModule/components/options/optionEditorItem';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'
import DndContext from '../../../../../common/components/dndContext';
import TestBackend from 'react-dnd-test-backend';
import { DragDropContext } from 'react-dnd';

function wrapInTestContext(DecoratedComponent, props) {
  return DragDropContext(TestBackend)(
    class TestContextContainer extends Component {
      render() {
        return <DecoratedComponent {...props} />;
      }
    }
  );
}

describe('Option editor item component', () => {
  const initialState  = {
    questions: {
      activeQuestion: {
        questionId: 1,
        questionType: 'CloseQuestion'
      }
    },
    options: {
      options: []
    },
    ui: {
      draggingElement: false
    }
  };

  const mockStore = configureStore();
  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(
      <DndContext>
        <OptionEditorItem store={store}/>
      </DndContext>
    );
    expect(wrapper).toBeDefined();
  })
  it('should render correctly', () => {
      const tree = shallow(
        <DndContext>
          <Snapshot
            questionType='CloseQuestion'
            questionId={1}
            option={{optionId: 1, title: 'foo'}}
          />
        </DndContext>
      );
      expect(toJson(tree)).toMatchSnapshot();
  });

  it('should call synchronizeOption', () => {
    const synchronizeOption = jest.fn();
    const wrapper = shallow(
      <Snapshot
        synchronizeOption={synchronizeOption}
        option={{optionId: 1, title: 'foo'}}
        questionType='CloseQuestion'
        questionId={1}
      />, DragDropContext
    );
    wrapper.instance().updateOption({title: 'foo', questionType: 'CloseQuestion', optionType: 'NormalOption'});
    expect(synchronizeOption.mock.calls.length).toEqual(1);
  })
});
