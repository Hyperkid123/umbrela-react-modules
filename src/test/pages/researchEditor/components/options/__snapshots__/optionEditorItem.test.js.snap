// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option editor item component should render correctly 1`] = `
<DNDContext>
  <OptionEditorItem
    option={
      Object {
        "optionId": 1,
        "title": "foo",
      }
    }
    questionId={1}
    questionType="CloseQuestion"
  />
</DNDContext>
`;
